activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -23,9 +23,7 @@ module ActiveRecord
   module ConnectionHandling # :nodoc:
     # Establishes a connection to the database that's used by all Active Record objects
     def postgresql_connection(config)
-      conn_params = config.symbolize_keys
-
-      conn_params.delete_if { |_, v| v.nil? }
+      conn_params = config.symbolize_keys.compact!
```

Comment:
@dhh can’t do an inline call since compact! returns nil if there’s no modifications. Can’t remember if compact has been added to hashes or if that’s still an AS core extension, but that would work.

Reply:
Oh, compact is on hashes now, that’ll work.

Reply:
I'm not sure if it matters here, but `compact` performs an extra allocation.

Reply:
Here is `compact!` (with bang)

Reply:
But we cannot use it inline (as @dhh suggested), because `compact!` will return `nil` if no changes were made

Reply:
Is this in a hot path? If not, let's just go with #compact.

Reply:
It is not a hot path (this method is only called, and thus extra hash allocation is made, for every newly created connection).
So updated with inline `compact` call.